embedding_reader/piece_builder.py

```diff
@@ -25,10 +38,13 @@ def build_pieces(headers, batch_size, start, end, max_piece_size=100000, metadat
     items = [filecount(*args) for args in zip(*[headers[col] for col in columns])]
 
     header_i = 0
-    while header_i < len(items) and items[header_i].count_before + items[header_i].count < start:
+    while header_i < len(items) and items[header_i].count_before + items[header_i].count <= start:
```

Comment:
That is correct indeed, thanks!
It's surprising I didn't hit that case. I guess it's pretty rare

Reply:
Yes, I believe that this case is really rare. Since last time we had an issue with an edge case like this in autofaiss, I checked very carefully for every `"<"`, `">"`. ;)